File: Chapter 07/Python/section_7_4_event_code/shared/event.py
from abc import ABC
from datetime import datetime
import uuid

class Event(ABC):
    """
    THE VIP BADGE: Abstract base class representing architectural constraints for events.
    
    By forcing all events to inherit from this base class, we ensure that every
    message moving through our pipeline has a standard shape. The infrastructure
    can rely on every event having an ID, a timestamp, and a correlation ID, 
    even if it doesn't understand the specific data payload.
    """
    event_id: uuid.UUID
    correlation_id: uuid.UUID
    occurred_on: datetime

File: Chapter 07/JS/section_7_4_event_code/shared/event.js
import crypto from 'crypto';

/**
 * THE VIP BADGE: Base class representing architectural constraints for events.
 * By forcing all events to inherit from this base class, we ensure that every
 * message moving through our reactive pipeline has a standard shape.
 */
export class Event {
    /**
     * Creates a standard base event.
     * @param correlationId - The ID used to trace the full request lifecycle.
     */
    constructor(correlationId) {
        this.eventId = crypto.randomUUID();
        this.correlationId = correlationId || crypto.randomUUID();
        this.occurredOn = new Date().toISOString();
    }
}